src/js_native_api_v8.cc

```diff
@@ -57,8 +57,28 @@
     (out) = v8::type::New((buffer), (byte_offset), (length));                  \
   } while (0)
 
-namespace v8impl {
+void napi_env__::InvokeFinalizerFromGC(v8impl::RefTracker* finalizer) {
+  if (module_api_version != NAPI_VERSION_EXPERIMENTAL) {
+    EnqueueFinalizer(finalizer);
+  } else {
+    // The experimental code calls finalizers immediately to release native
+    // objects as soon as possible, but it suspends use of JS from finalizer.
+    // If JS calls are needed, then the finalizer code must call
+    // node_api_post_finalizer.
+    if (last_error.error_code == napi_ok && last_exception.IsEmpty()) {
+      bool saved_suspend_call_into_js = suspend_call_into_js;
+      finalizer->Finalize();
+      suspend_call_into_js = saved_suspend_call_into_js;
```

Comment:
The comment says calling finalizer suspends use of JS from finalizer, but I didn't see where `suspend_call_into_js` is set to `true`?

Reply:
Oops... Good catch! Thank you!